pkg/controller/component/helper.go

```diff
@@ -144,3 +153,98 @@ func syncConfigMap(
 	}
 	return cm, cmHash, nil
 }
+
+func getContainerImage(
+	extender extenderutil.UnstructuredExtender,
+	obj *unstructured.Unstructured,
+	containerName string) string {
+	if obj == nil {
+		return ""
+	}
+	containers := extender.GetContainers(obj)
+	for _, ctr := range containers {
+		if ctr["name"] == containerName {
+			return ctr["image"].(string)
+		}
+	}
+	return ""
+}
+
+func isUpdating(
+	component v1alpha1.NebulaClusterComponentter,
+	podClient kube.Pod,
+	obj *unstructured.Unstructured) (bool, error) {
+	extender := extenderutil.New()
+	if extenderutil.IsUpdating(extender, obj) {
+		return true, nil
+	}
+
+	selector, err := label.Label(component.GenerateLabels()).Selector()
+	if err != nil {
+		return false, err
+	}
+
+	pods, err := podClient.ListPods(component.GetNamespace(), selector)
+	if err != nil {
+		return false, fmt.Errorf(
+			"failed to get pods for cluster %s/%s, selector %s, error: %s",
+			component.GetNamespace(),
+			component.GetClusterName(),
+			selector,
+			err,
+		)
+	}
+	for i := range pods {
+		pod := pods[i]
+		revisionHash, exist := pod.Labels[appsv1.ControllerRevisionHashLabelKey]
+		if !exist {
+			return false, nil
+		}
+		if revisionHash != component.GetUpdateRevision() {
+			return true, nil
+		}
+	}
+	return false, nil
```

Comment:
Can you ensure that all pods meet `revisionHash != component.GetUpdateRevision()`?

Reply:
Yes